Webhook doesn't work without a password/secret

Gogs Version (Docker Tag): 0.10.8
Jenkins Plugin Version: 1.0.9
Reference: #https://github.com/gogits/gogs/issues/4269

Hello,

I'm forced to use a password/secret when I want to use Webhooks of Gogs, otherwise the Jenkins plugin (1.0.9) throws an exception:
<pre style="margin:2em; clear:both">javax.servlet.ServletException: net.sf.json.JSONException: JSONObject["secret"] not found.
....
Caused by: net.sf.json.JSONException: JSONObject["secret"] not found.
	at net.sf.json.JSONObject.getString(JSONObject.java:2040)
	at org.jenkinsci.plugins.gogs.GogsWebHook.doIndex(GogsWebHook.java:163)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.kohsuke.stapler.Function$InstanceFunction.invoke(Function.java:335)
	at org.kohsuke.stapler.Function.bindAndInvoke(Function.java:175)
	at org.kohsuke.stapler.Function.bindAndInvokeAndServeResponse(Function.java:108)
	at org.kohsuke.stapler.IndexDispatcher.dispatch(IndexDispatcher.java:26)
	at org.kohsuke.stapler.Stapler.tryInvoke(Stapler.java:746)
	... 60 more
</pre>

Since Gogs 0.10.x there is no JSON field "secret" anymore, only the header field "X-Gogs-Signature" will be send, which holds the encrypted secret or it is just empty. I guess the back-support for Gogs version 0.9.x and less causes the issue. 

Anyway, older versions of "Gogs 0.9.x" are working fine together with the Jenkins plugin "1.0.9" or less. But I would appreciate, if you could fix this issue. Or is there any additional configuration I have to do?

--- a/src/main/java/org/jenkinsci/plugins/gogs/GogsWebHook.java
+++ b/src/main/java/org/jenkinsci/plugins/gogs/GogsWebHook.java
@@ -31,20 +31,20 @@ import jenkins.model.Jenkins;
 import net.sf.json.JSONObject;
 import org.acegisecurity.context.SecurityContext;
 import org.acegisecurity.context.SecurityContextHolder;
+import org.apache.commons.codec.binary.Hex;
 import org.apache.commons.io.IOUtils;
 import org.kohsuke.stapler.StaplerRequest;
 import org.kohsuke.stapler.StaplerResponse;
 
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
 import java.io.IOException;
 import java.io.UnsupportedEncodingException;
 import java.net.URLDecoder;
+import java.nio.charset.Charset;
 import java.util.LinkedHashMap;
 import java.util.Map;
 import java.util.logging.Logger;
-import javax.crypto.Mac;
-import javax.crypto.spec.SecretKeySpec;
-import java.nio.charset.Charset;
-import org.apache.commons.codec.binary.Hex;
 
 /**
  * @author Alexander Verhaar
@@ -160,7 +160,7 @@ public class GogsWebHook implements UnprotectedRootAction {
 
         String gSecret = null;
         if (gogsSignature==null){
-          gSecret = jsonObject.getString("secret");  /* Secret provided by Gogs    */
+          gSecret = jsonObject.optString("secret", null);  /* Secret provided by Gogs < 0.10.x   */
         }
         else{
           try{

